tests/sql/column/test_column_select_lateral_alias_ref.py

```diff
@@ -27,6 +36,150 @@ def test_column_top_level_lateral_ref():
                 ColumnQualifierTuple("id", "public.tgt_tbl1"),
             ),
         ],
+        test_sqlparse=False,
```

Comment:
revisiting the test case (test_column_top_level_lateral_ref) from last commit:
```sql
insert into public.tgt_tbl1
    select
        name               as user_name,
        user_name || email as id         -- lateral ref
    from
        public.src_tbl1
```

Column lineage `name -> user_name` and `email -> id` are correct.

But `name -> id` is the one I'm in doubt, because:

1. In dialect that doesn't support lateral column alias reference, it should be `user_name -> id`. (If user_name doesn't exist in source table, then database will throw an error, but sqllineage doesn't check this, just like select any columns from a table, we don't check if that column exists)
2. In dialect that supports this feature (redshift, spark 3.4+), we need to check first if column exists in the table, if yes then it's  `user_name -> id`, otherwise fallback to previous alias and output `name -> id`

So I'd prefer we output `name -> id` by default.

Reply:
According to your meaning, should the last sentence be:

So I'd prefer we output **user_name -> id** by default.

Reply:
Yes, it's a typo. Sorry for the confusion. I just updated the comment.

Reply:
1. Dialect plus version determine whether to enable lateral alias ref
2. Add configuration to let users decide whether to enable it.

Which of the above two methods is more elegant? Or is there any other more elegant method?

Reply:
However, the current context only has dialect and no database version number information.

Reply:
An explicit LATERAL_COLUMN_ALIAS_REFERENCE config with default value as False is preferred. 

Taking sparksql as example, as you mentioned we don't know if user is using 3.4+ or not.

Reply:
ok I'll take care of it